app/src/main/java/com/raulh82vlc/functionalkotlinarrow/ui/MainActivity.kt

```diff
@@ -16,14 +16,60 @@
 
 package com.raulh82vlc.functionalkotlinarrow.ui
 
-import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import android.widget.Toast
+import arrow.effects.fix
 import com.raulh82vlc.functionalkotlinarrow.R
+import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCacheModel
+import com.raulh82vlc.functionalkotlinarrow.presentation.FeedListView
+import com.raulh82vlc.functionalkotlinarrow.presentation.Presentation
+import com.raulh82vlc.functionalkotlinarrow.ui.adapter.FeedAdapter
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), FeedListView {
+    private lateinit var feedAdapter: FeedAdapter
```

Comment:
Humongous NO! Don't use `lateinit` if you can avoid it. This can be done more easily with `by lazy {  }`.

Reply:
honestly, I need to read more about differences between both, I got used to do this with **lateinit** all the time, why does it have a problem?

Reply:
Actually, in this case, since the adapter is actually not dependent on the activity it could just be instantiated with the property declaration instead of in `onCreate`. Same as in Java 😄

Reply:
done